packages/block-library/src/post-template/index.php

```diff
@@ -86,3 +86,64 @@ function register_block_core_post_template() {
 	);
 }
 add_action( 'init', 'register_block_core_post_template' );
+
+
+/**
+ * Renders the legacy `core/query-loop` block on the server.
+ * It triggers a developer warning and then calls the renamed
+ * block's `render_callback` function output.
+ *
+ * This can be removed when WordPress 5.9 is released.
+ *
+ * @param array    $attributes Block attributes.
+ * @param string   $content    Block default content.
+ * @param WP_Block $block      Block instance.
+ *
+ * @return string Returns the output of the query, structured using the layout defined by the block's inner blocks.
+ */
+function render_legacy_query_loop_block( $attributes, $content, $block ) {
+	trigger_error(
+		/* translators: %1$s: Block type */
+		sprintf( __( 'Block %1$s has been renamed to Post Template. %1$s will be supported until WordPress version 5.9.', 'gutenberg' ), $block->name ),
+		headers_sent() || WP_DEBUG ? E_USER_WARNING : E_USER_NOTICE
+	);
+	return render_block_core_post_template( $attributes, $content, $block );
+}
+
+/**
+ * Complements the renaming of `Query Loop` to `Post Template`.
+ * This ensures backwards compatibility for any users running the Gutenberg
+ * plugin who have used Query Loop prior to its renaming.
+ *
+ * This can be removed when WordPress 5.9 is released.
+ *
+ * @see https://github.com/WordPress/gutenberg/pull/32514
+ */
+function gutenberg_register_legacy_query_loop_block() {
+	$registry = WP_Block_Type_Registry::get_instance();
+	if ( $registry->is_registered( 'core/query-loop' ) ) {
+		unregister_block_type( 'core/query-loop' );
+	}
+	register_block_type(
+		'core/query-loop',
+		array(
+			'category'          => 'design',
+			'uses_context'      => array(
+				'queryId',
+				'query',
+				'queryContext',
+				'displayLayout',
+				'templateSlug',
+			),
+			'supports'          => array(
+				'reusable' => false,
+				'html'     => false,
+				'align'    => true,
+			),
+			'style'             => 'wp-block-post-template',
+			'render_callback'   => 'render_legacy_query_loop_block',
+			'skip_inner_blocks' => true,
+		)
+	);
+}
+add_action( 'wp_loaded', 'gutenberg_register_legacy_query_loop_block' );
```

Comment:
I used `wp_loaded` so that `Query Loop` has been registered from 'core'.

Reply:
not sure I understand, why not just init?

Reply:
Hm... I must had something different while coding this.. I tested with `init` and it seems to work just fine..